Style changes in the create/edit patients screens

matrix-amiel opened the pull request:
Hi.

The new bootstrap style in the new patient form and demographics form it's great but the header is very big and take too place in the screen. I think..
I did minor changes in the style of header and now a user experience is better.

Hopefully you love it.

Thanks 
amiel  

![image](https://user-images.githubusercontent.com/17809866/29168772-154f6442-7dd9-11e7-95a7-ed2e7fd522ea.png)

bradymiller commented:
very cool to see @zbig01 's new theme in this screenshot!!

bradymiller commented:
@robertdown and @zbig01 ,

Was hoping to get your thoughts on this approach. There are several layouts that do this (such as also at Administration->Users)

It also seems like the title takes up lots of space in the current codebase for these. Options are:
1. change title from h1/h2 to h3
2. remove title altogether since it is on the tab (note this could be an issue for frames layout users though)
3. migrate the buttons to the right as matrix-amiel as done
4. guessing lots of other possibilities

thoughts?

-brady

zbig01 commented:
I have used h2 in all the forms that I restyled. It does take up less space that h1. I think leaving it in is a good idea as the title gives a clear idea of purpose of the page. You appreciate its need when you see three locked tabs - the pages are skewed and do not lie under the tab, the title is then an unambiguous indicator of the purpose of the page. Also when multiple tabs are open a dynamic title like my changes to Fees > Fee Sheet, Encounter > Administrative > Procedure Orders, Fees > Payment,and Fees Checkout brings further clarity to the pages.

 I  think h3 looks a bit too small.

Please be aware that most of that space is used by the page-header class that has 40px top and 20 px bottom margin.  Maybe time for some oe customization :). When you look carefully at bootstrap you will note a lot of padding and margins. I noted that  it caused problems in displaying data in input text boxes and in select boxes forcing you to use a larger col size for these boxes.


zbig01


________________________________
From: Brady Miller <notifications@github.com>
Sent: Friday, August 11, 2017 12:02 AM
To: openemr/openemr
Cc: zbig01; Mention
Subject: Re: [openemr/openemr] Style changes in the create/edit patients screens (#1003)


@robertdown<https://github.com/robertdown> and @zbig01<https://github.com/zbig01> ,

Was hoping to get your thoughts on this approach. There are several layouts that do this (such as also at Administration->Users)

It also seems like the title takes up lots of space in the current codebase for these. Options are:

  1.  change title from h1/h2 to h3
  2.  remove title altogether since it is on the tab (note this could be an issue for frames layout users though)
  3.  migrate the buttons to the right as matrix-amiel as done
  4.  guessing lots of other possibilities

thoughts?

-brady

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/openemr/openemr/pull/1003#issuecomment-321743933>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AMEW11nJVSnylnBXSz1LVtDdjxoTK1beks5sW_yQgaJpZM4OzTV7>.

matrix-amiel commented:
Hi.
Thanks for your comments.
I changed all the titles to h2 tag and I removed from 'page-header' class a large padding.
I tested it in some pages as I could and it seem good.

Hopefully it's better now.
Thanks

Amiel

robertdown commented on interface/new/new_comprehensive.php:
```diff
@@ -416,24 +416,25 @@ function searchme() {
-            <div class="col-xs-12">
-                <div class="page-header">
-                    <h1><?php echo xlt('Search or Add Patient');?></h1>
+        <div class="page-header">
```
Recommend only displaying this if the user is using Left side Navigation (Not tabs) - the title is already displayed on the tabs bar just above this, it's a bit redundant in this particular case.

matrix-amiel replied:
I didn't add a titles. I only made the header nicer.
This idea could be in new PR.

robertdown replied:
Since we're working on the code in this PR and it's relavent still suggest we do it now

robertdown commented on interface/new/new_comprehensive.php:
```diff
@@ -416,24 +416,25 @@ function searchme() {
-                        <button type="button" class="btn btn-default btn-search" id="search" value="<?php echo xla('Search'); ?>">
-                            <?php echo xlt('Search'); ?>
+
+                <div class="col-xs-12 col-sm-6 col-md-4">
```
This needs to go below the form, preferably inline with the left-most input field, though that may not be super feasible in this case.

matrix-amiel replied:
There is 'Create' button in the header it must be into the form.

robertdown commented on interface/patient_file/summary/demographics_full.php:
```diff
@@ -426,22 +426,21 @@ function policykeyup(e) {
-            <div class="col-xs-12">
-                <div class="page-header">
-                    <h1><?php echo xlt('Current Patient');?></h1>
+        <div class="page-header">
```
Put the `page-header` inside the `col` div

matrix-amiel replied:
Why? it only for marging-top-bottom.

robertdown commented on interface/patient_file/summary/demographics_full.php:
```diff
@@ -426,22 +426,21 @@ function policykeyup(e) {
-                        <?php echo xlt('Cancel'); ?>
-                    </a>
+                <div class="col-xs-12 col-sm-6 col-md-4">
+                    <div class="btn-group btn-header">
```
What's the `btn-header` for?

matrix-amiel replied:
move button to straight line line with the title

robertdown commented on interface/patient_file/transaction/add_transaction.php:
```diff
@@ -332,22 +332,21 @@ function submitme() {
-            <div class="col-xs-12">
-                <div class="page-header">
-                    <h1><?php echo xlt('Add/Edit Patient Transaction');?></h1>
+        <div class="page-header">
```
Move `page-header` inside `col` div

matrix-amiel replied:
Why? it only for marging-top-bottom.

robertdown commented on interface/themes/core.css:
```diff
@@ -100,3 +100,7 @@ button[href*="deleter"]:before {
 .btn-graph::before {
     content: '\f201\00a0\00a0';
 }
+.page-header{
```
Remove this from core, it's a theme-specific issue and recommend themes not even bothering with it

robertdown commented on interface/themes/style_babyblu.css:
```diff
@@ -1356,3 +1356,7 @@ table.dataTable#therapy_groups_list tbody td{
     color: #444444 !important;
 }
+
+.btn-header{
```
Remove this class all together

zbig01 commented:
Hi @matrix-amiel

Looking through your code I note the following:
    - The page is not consistently using bootstrap
    - There is a class="container-fluid" div **inside** the form that contains only the page header and search and Create New patient button
    - The elements of the form are in a table  based layout
    - The table is not having a class='table' div
    - There is a lot of the origianl inline styling left in place
    
May I suggest that you look at some of the pages that @robertdown has modified for some ideas. Especially the Procedure > Providers page that is fully responsive.  Also look at the PR# 993 where we are having an ongoing discussion about other pages that have been restyled.

Cheers

matrix-amiel commented:
O.k.
I just wanted to do minor fix because current UI is looks very bad. so I left it I don't have time to fix all the screens.
